src/core.js

```diff
@@ -392,6 +393,99 @@ define([
         ];
     };
 
+    fn._init_bulk_update_questions = function () {
+        var _this = this,
+            menuItems = this.getBulkUpdateMenuItems();
+
+        var $lastItem = this.$f.find('.fd-bulk-update-header');
+        $lastItem.nextUntil(".divider").remove();
+        _(menuItems).each(function (menuItem) {
+            var $menuLink = $(util.format(
+                    "<a tabindex='-1' href='#'>{name}</a>",
+                    _.extend(menuItem)
+                )).click(function (e) {
+                    e.preventDefault();
+                    _this.showConfirmBulkActionModal(menuItem.confirmMessage, menuItem.confirmAction);
+                }),
+                $newItem = $("<li></li>").append($menuLink);
+            $lastItem.after($newItem);
+            $lastItem = $newItem;
+        });
+
+    };
+
+    fn.getBulkUpdateMenuItems = function () {
+        var _this = this,
+            menuItems =  [
+                {
+                    name: gettext('Make Required'),
+                    confirmMessage: gettext("You are about to make all existing " +
+                        "questions on this form required. This action will overwrite " +
+                        "the current configuration of this form and can only be " +
+                        "undone by editing individual questions."),
```

Comment:
Would it be hard to integrate these actions with the undo manager? It would make a far better user experience if they could be undone like other modifications (and could omit the modal prompt).

Reply:
I'm not super familiar with the undo manager. so from my perspective that seems like a huge ask given my uncertainty level and should be considered as a separate ask from product unless Dev is ok with an additional delay.

Reply:
I do like this idea, but note that currently we present the Undo option **only** on question deletions, so we will introducing a new UX pattern if we support Undo for form updates that not question deletions. Worth considering if this could have any drawbacks. Would love to hear your thoughts on this @millerdev, and also @orangejenny and @biyeun.

Reply:
Fair enough. This could be spun off as a separate effort too if we wanted to ship the feature and then improve it later.

Reply:
We will 100% address this during UAT.